barretenberg/cpp/src/barretenberg/vm2/constraining/relations/sha256.test.cpp

```diff
@@ -0,0 +1,74 @@
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <cstdint>
+
+#include "barretenberg/vm2/constraining/testing/check_relation.hpp"
+#include "barretenberg/vm2/generated/flavor_settings.hpp"
+#include "barretenberg/vm2/generated/relations/sha256.hpp"
+#include "barretenberg/vm2/simulation/events/event_emitter.hpp"
+#include "barretenberg/vm2/simulation/memory.hpp"
+#include "barretenberg/vm2/testing/macros.hpp"
+#include "barretenberg/vm2/tracegen/sha256_trace.hpp"
+// TMP
+#include "barretenberg/vm2/common/memory_types.hpp"
+#include "barretenberg/vm2/simulation/events/memory_event.hpp"
+#include "barretenberg/vm2/simulation/sha256_compression.hpp"
+#include "barretenberg/vm2/simulation/testing/mock_context.hpp"
+#include "barretenberg/vm2/tracegen/test_trace_container.hpp"
+
+namespace bb::avm2::constraining {
+namespace {
+
+using ::testing::ReturnRef;
+using ::testing::StrictMock;
+using tracegen::TestTraceContainer;
+using FF = AvmFlavorSettings::FF;
+using C = Column;
+using sha256 = bb::avm2::sha256<FF>;
+
+// This test imports a bunch of external code since hand-generating the sha256 trace is a bit laborious atm.
```

Comment:
Please add an "empty row test": just a single row with `precomputed_first_row = 1` and run the whole relation. This is to check that the relation passes on empty rows (and kind of test skippable). See David's PR.